### ISSUE TITLE
Clean up player objects in tests?

### ISSUE THREAD

**Issue by chemoish** [videojs/video.js#2895]
Many tests do not call `player.dispose()`, not sure if this matters?

https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L23

```js
// correct
test('should create player instance that inherits from component and dispose it', function(){
  var player = TestHelpers.makePlayer();

  ok(player.el().nodeName === 'DIV');
  ok(player.on, 'component function exists');

  player.dispose();
  ok(player.el() === null, 'element disposed');
});
```

https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L176

```js
// incorrect?
test('should use an class name that begins with an alpha character', function(){
  let alphaPlayer = TestHelpers.makePlayer({ id: 'alpha1' });
  let numericPlayer = TestHelpers.makePlayer({ id: '1numeric' });

  let getStyleText = function(styleEl){
    return (styleEl.styleSheet && styleEl.styleSheet.cssText) || styleEl.innerHTML;
  };

  alphaPlayer.width(100);
  numericPlayer.width(100);

  ok(/\s*\.alpha1-dimensions\s*\{/.test(getStyleText(alphaPlayer.styleEl_)), 'appends -dimensions to an alpha player ID');
  ok(/\s*\.dimensions-1numeric\s*\{/.test(getStyleText(numericPlayer.styleEl_)), 'prepends dimensions- to a numeric player ID');
});
```

---

Non-exhaustive list:

https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L129
https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L175
https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L235
https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L466
https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L481
https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L510
https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L526
https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L546
https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L596
https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L631
https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L647
https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L666
https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L699
https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L713
https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L727
https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L735
https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L743
https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L761
https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L791
https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L823
https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L838
https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L848

**Comment by gkatsev** [comment]
If my labels weren't clear, yeah we probably should make sure all those players are disposed off properly :)

**Comment by gkatsev** [comment]
I think this is fixed via #3524.
